src/logic/main.js

```diff
@@ -51,11 +54,101 @@ function isMoveLegal() {
     return true;
 }
 
+
+// ++++++++++++++++++BOT ALGORITHM ++++++++++++++++++++++
+
 function playBotMove() {
     GameState.currentPlayer = PlayerEnum.Bot;
-    return playGameMove(GameState.BotPile.cards[0].id);
+    let leadingCard = GameState.leadingCard;
+    let selectedCard;
+    let actionState = GameState.actionState;
+    let bot = GameState[PileTypeEnum.BotPile];
+    let matchedCard;
+    debugger;
+    // 4.1 if actionState is twoPlusInvoked and bot has twoPlus Card - mark it as selectedCard.
+    if (actionState === `${CardActionEnum.TwoPlus}Invoked`) {
+        if (matchedCard = getCardInHand(bot, [{action: CardActionEnum.TwoPlus}])) {
+            selectedCard = matchedCard;
+
+        } // if twoPlusInvoked and bot doesn't have a two plus - mark the top card of the draw pile as the selectedCard.
+        else {
+            debugger;
+            selectedCard = GameState.DrawPile.getTop();
+        }
+    } // if actionState is takiInvoked and bot has a card with the same color of the leadingCard - mark it as selectedCard.
+    else if ( (actionState === `${CardActionEnum.Taki}Invoked`) &&
+            (matchedCard = getCardInHand(bot, [{color: leadingCard.color}])) ){
+            selectedCard = matchedCard;
+    } // ( if we got here no actionState was invoked)
+    else
```

Comment:
Since there is pretty much a lot of code in this function please consider of splitting it to more then one (for example the split can be active action or inactive)

Reply:
ok